File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/order/OrderCartProduct.kt
package com.sudo248.sudoo.domain.entity.order

import com.sudo248.base_android.base.ItemDiff

data class OrderCartProduct(
    val cartProductId: String,
    val cartId: String,
    val quantity: Int,
    val totalPrice: Double,
    val product: OrderProductInfo,
    val purchasePrice: Double? = null,
): ItemDiff {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is OrderCartProduct && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is OrderCartProduct && other.cartProductId == cartProductId
    }

}

File: client/web/sudoo/lib/domain/model/discovery/order_cart_product.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/discovery/order_product_info.dart';

part 'order_cart_product.g.dart';

@JsonSerializable(explicitToJson: true)
class OrderCartProduct {
  final String cartProductId;
  final String cartId;
  final int quantity;
  final double totalPrice;
  final OrderProductInfo product;

  OrderCartProduct(
    this.cartProductId,
    this.cartId,
    this.quantity,
    this.totalPrice,
    this.product,
  );

  factory OrderCartProduct.fromJson(Map<String, dynamic> json) => _$OrderCartProductFromJson(json);

  Map<String, dynamic> toJson() => _$OrderCartProductToJson(this);
}
